src/System.Net.Sockets/src/System/Net/Sockets/_ReceiveMessageOverlappedAsyncResult.Windows.cs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft. All rights reserved.
```

Comment:
The `Overlapped` bit isn't really interesting here. The `AsyncResult` bit, however, is very relevant: these `AsyncResult` types and the code that interacts with them have particular semantics on Windows that are most easily matched on *nix by retaining the various subtypes.